src/main/scala/org/broadinstitute/dsde/firecloud/service/UserService.scala

```diff
@@ -114,31 +119,45 @@ trait UserService extends HttpService with PerRequestCreator with FireCloudReque
     } ~
     requireUserInfo() { userInfo =>
     pathPrefix("api") {
-      path("profile" / "billing") { requestContext =>
-        val extReq = Get(UserService.billingUrl)
-        externalHttpPerRequest(requestContext, extReq)
+      path("profile" / "billing") {
+        passthrough(UserService.billingUrl, HttpMethods.GET)
+      } ~
+      path("profile" / "billingAccounts") {
+        hostName { callbackHost => requestContext =>
+          val pipeline = authHeaders(requestContext) ~> sendReceive
+          val extReq = Get(UserService.billingAccountsUrl)
+          pipeline(extReq) onComplete {
+            case Success(response) if response.status == OK => requestContext.complete(OK, response.entity)
+            case Success(response) if response.status == Forbidden =>
+              // user does not have appropriate scopes.  Ask user to enable them via OAuth and then redirect back here
+
+              val callbackPath = "api/profile/billingAccounts"
+              val stateForCallback = s"$callbackHost#$callbackPath"
+              val scopes = Option(HttpGoogleServicesDAO.billingAdminScopes)
+
+              oauthUtil.initiateAuth(stateForCallback, "auto", requestContext, scopes)
+            case _ => respondWithUserInfo(InternalServerError, InternalServerError.defaultMessage, requestContext)
```

Comment:
Current version of the Rawls endpoint has only `OK` and `Forbidden` as valid states.  Remember to update this if the final endpoint changes.